File: go/p1_3_rand_squares.go
package main

import . "g2d"

func main() {
    w, h := 640, 480
    side := 100

    n := ToInt(Prompt("N? "))

    InitCanvas(Size{w, h})
    i := 0
    for i < n {
        r, g, b := RandInt(0, 255), RandInt(0, 255), RandInt(0, 255)
        x, y := RandInt(0, w - side), RandInt(0, h - side)
        SetColor(Color{r, g, b})
        FillRect(Rect{x, y, side, side})
        i += 1
    }
    MainLoop(nil, 0)
}

File: exercises/p1_3_rand_squares.py
#!/usr/bin/env python3
'''
@author Michele Tomaiuolo - http://www.ce.unipr.it/people/tomamic
@license This software is free - http://www.gnu.org/licenses/gpl.html

Quadrati casuali
- Chiedere all'utente un numero n
- Disegnare n quadrati
-- Tutti con lato di 100 pixel
-- Ciascuno in posizione casuale
-- Ciascuno con un colore casuale
'''

import game2d as g2d
from random import randrange

W, H = 640, 480
SIDE = 100

n = int(input("n? "))

g2d.init_canvas((W, H))

i = 0
while i < n:
    color = randrange(255), randrange(255), randrange(255)
    x, y = randrange(W - SIDE), randrange(H - SIDE)
    g2d.draw_rect(color, (x, y, SIDE, SIDE))
    i += 1

g2d.main_loop()
